### PR TITLE
test: improve test coverage

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

Add a few test cases, also removed code branches that can never be reached given proper boundary validation..

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: packages/docusaurus-types/src/index.d.ts [hunk 1]
```diff
@@ -161,10 +161,7 @@ export type ImportedPresetModule = PresetModule & {
   default?: PresetModule;
 };
 
-export type PresetConfig =
-  | [string, Record<string, unknown>]
-  | [string]
```
Comment (Josh-Cena) [inline packages/docusaurus-types/src/index.d.ts:166]: This type has been removed, because it never works during validation
